### PR TITLE
Use PSR6 cache

### LINKED ISSUE #390: Use PSR-6 cache
Once #389 is merged we should remove our custom cache layer and replace it with PSR-6.

### PR DESCRIPTION
**Pull request by Nyholm**
This PR removes our caching layer an introduce PSR-6 cache instead. 

(It feels great to make a PR like this. Remove 500 lines just to add 17) =)

This will fix #390

This is BC break so it should be done before we tag 2.0.

### CODE REVIEW

File: lib/Github/Client.php [hunk 1]
```diff
@@ -374,15 +374,24 @@ public function addHeaders(array $headers)
     }
 
     /**
+     * Add a cache plugin to cache responses locally.
      * @param CacheItemPoolInterface $cache
      */
-    public function useCache(CacheItemPoolInterface $cachePool)
+    public function addCache(CacheItemPoolInterface $cachePool)
     {
         $this->removePlugin(Plugin\CachePlugin::class);
```
Comment (GrahamCampbell) [inline lib/Github/Client.php]: Maybe call the removeCache() method here?

**Comment by Nyholm** [comment]
The removeCache() function will not work properly until #405 is merged.

File: lib/Github/Client.php (hunk 1 above)
Comment (Nyholm) [inline lib/Github/Client.php]: I was just a 10 seconds slower than you. It is added now

File: composer.json [hunk 2]
```diff
@@ -18,10 +18,13 @@
     ],
     "require": {
         "php":             "^5.5|^7.0",
```
Comment (GrahamCampbell) [inline composer.json]: We can probs fix the indentation here. :)

**Comment by GrahamCampbell** [comment]
Looks good to go. 👍

**Comment by Nyholm** [comment]
Thank you for your reviews!

**Pull request merged**